src/Cli/dotnet/commands/dotnet-test/VSTestArgumentConverter.cs

```diff
@@ -148,14 +177,39 @@ public List<string> Convert(string[] args, out List<string> ignoredArgs)
                     {
                         UpdateVerbosity(arg, newArgList);
                     }
-                    else if (IgnoredArguments.Contains(activeArgument))
+                    // This case should be processed before 's_ignoredArguments.Contains'.
+                    else if (s_switchArguments.Contains(activeArgument))
+                    {
+                        if (s_ignoredArguments.Contains(activeArgument))
+                        {
+                            ignoredArgs.Add(activeArgument);
+                        }
+                        else
+                        {
+                            newArgList.Add(activeArgument);
+                        }
+
+                        // The only real case where we would end-up here is when the item under test (.dll or .exe)
+                        // is passed as last argument so we could potentially check the extension and for invalid ones,
+                        // either add it to the ignored args or have a specific failure.
+                        newArgList.Add(arg);
+                    }
+                    else if (s_ignoredArguments.Contains(activeArgument))
                     {
                         ignoredArgs.Add(activeArgument);
                         ignoredArgs.Add(arg);
                     }
-                    else if (blame.IsBlameArg(activeArgument, arg))
+                    else if (BlameArgs.IsBlameArg(activeArgument))
                     {
-                        blame.UpdateBlame(activeArgument, arg);
+                        if (!BlameArgs.IsBlameSwitch(activeArgument)
+                            || BlameArgs.IsLegacyBlame(activeArgument, arg))
+                        {
+                            blame.UpdateBlame(activeArgument, arg);
+                        }
+                        else
+                        {
+                            newArgList.Add(arg);
+                        }
```

Comment:
Main change to avoid swallowing the arg when active argument is a switch (blame switch here).